main_test.go

```diff
@@ -153,3 +156,60 @@ func TestIntegrationScripts(t *testing.T) {
 		fmt.Printf("TAP: # Total tests: %d\n", totalTapTests)
 	}
 }
+
+func TestStartPreprocess(t *testing.T) {
+	// temp dir
+	tempDir := os.TempDir()
+
+	// create temp file with garantee
+	// wrong file body
+	tempFile, err := newTempFile(tempDir, "c2go", "preprocess.c")
+	if err != nil {
+		t.Errorf("Cannot create temp file for execute test")
+	}
+	defer os.Remove(tempFile.Name())
+
+	fmt.Fprintf(tempFile, "#include <AbsoluteWrongInclude.h>\nint main(void){\nwrong();\n}")
+
+	err = tempFile.Close()
+	if err != nil {
+		t.Errorf("Cannot close the temp file")
+	}
+
+	var args ProgramArgs
+	args.inputFile = tempFile.Name()
+
+	err = Start(args)
+	if err == nil {
+		t.Errorf("Cannot test preprocess of application")
+	}
+}
+
+func TestGoPath(t *testing.T) {
+	gopath := "GOPATH"
+
+	existEnv := os.Getenv(gopath)
+	if existEnv == "" {
+		t.Errorf("$GOPATH is not set")
+	}
+
+	// return env.var.
+	defer func() {
+		err := os.Setenv(gopath, existEnv)
+		if err != nil {
+			t.Errorf("Cannot restore the value of $GOPATH")
+		}
+	}()
+
+	// reset value of env.var.
+	err := os.Setenv(gopath, "")
+	if err != nil {
+		t.Errorf("Cannot set value of $GOPATH")
+	}
+
+	// testing
+	err = Start(ProgramArgs{})
+	if err == nil {
```

Comment:
Thanks for noticing that the version is wrong. I have fixed it.

If you look at the code again:

```go
if err == nil {
    t.Errorf(err.Error())
}
```

If there is no error the test will panic because `err.Error()` is an impossible expression. If there is an error you are not checking what the error is. So if something goes wrong that is unrelated to the test, the test will still pass.